evm/sequencer/server.go

```diff
@@ -135,3 +111,27 @@ func (s *Sequencer) RegisterGameShard(
 	s.tq.AddInitMsg(req.GetNamespace(), req.GetRouterAddress())
 	return &shard.RegisterGameShardResponse{}, nil
 }
+
+// serverCallInterceptor catches calls to handlers and ensures they have the right secret routerKey.
+func (s *Sequencer) serverCallInterceptor(
+	ctx context.Context,
+	req any,
+	_ *grpc.UnaryServerInfo,
+	handler grpc.UnaryHandler,
+) (resp any, err error) {
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		return nil, status.Errorf(codes.Unauthenticated, "missing metadata")
+	}
+
+	routerKey, ok := md["router-routerKey"]
+	if !ok || len(routerKey) == 0 {
```

Comment:
nit: the ok check is redundant here. If the key is missing, len(routerKey) will be 0.

Reply:
addressed